internal/controllers/gateway/gateway_controller.go

```diff
@@ -583,6 +585,18 @@ func (r *GatewayReconciler) reconcileUnmanagedGateway(ctx context.Context, log l
 		debug(log, gateway, "Determining listener configurations from Kong data-plane")
 		kongListeners, err = r.determineListenersFromDataPlane(ctx, svc, kongListeners)
 		if err != nil {
+			errNoReadyClient := dataplane.NoReadyGatewayClientsError{}
+			if errors.As(err, &errNoReadyClient) {
+				// Requeue the request to reconcile the gateway again after `retryAfter` but not return the error to trigger a Reconcile Error
+				// when no gateway is available to fetch information of listeners.
+				// REVIEW: Should we implement some exponential backoff strategy here instead of a fixed retry delay?
```

Comment:
This will be handled by the controller-runtime framework if we do not set the `RequeueAfter` but only set the `Requeue`. I'd suggest we go this route.

Reply:
Changed like this.